tools/dashboard

```diff
@@ -7,6 +7,9 @@ cd "$(readlink -f "$(dirname "$0")")/.."
 
 if [ -n "${PROJECT}" -a "$1" != "auto" ]; then
   . config/options ""
+else
+  BUILD_ROOT=$(. config/options ""; echo ${BUILD_ROOT})
+  BUILD_BASE=$(. config/options ""; echo ${BUILD_BASE})
```

Comment:
If an invalid PROJECT or DEVICE or ARCH is passed to `config/options` it will `die()`, which you're not handling here in your sub-process.

```suggestion
  BUILD_BASE=$(. config/options ""; echo ${BUILD_BASE}) || exit
```
would handle the error, exiting immediately. The same change is needed for `BUILD_ROOT`.

However as `BUILD_ROOT` and `BUILD_BASE` are only ever used once, and together, you could get the path and build directory name with a single combined call to `config/options`:
```suggestion
  BUILD_PREFIX=$(. config/options "" && echo ${BUILD_ROOT}/${BUILD_BASE}) || exit
```
then...
```
-     for dir in $(ls -1td ${BUILD_ROOT}/${BUILD_BASE}.*/.threads 2>/dev/null); do
+     for dir in $(ls -1td ${BUILD_PREFIX}.*/.threads 2>/dev/null); do
```

Note that there's no need to handle `die()` in line 9 as we're sourcing `config/options` and the entire script will exit automatically.

Reply:
You could even forgo sourcing the additional `config/options` entirely, at which point the only change required to `tools/dashboard` is:
```
-     for dir in $(ls -1td "${PWD}"/build.*/.threads 2>/dev/null); do
+     for dir in $(ls -1td ${BUILD_DIR:-${PWD}}/build.*/.threads 2>/dev/null);
```